text/dependencies/rfcs/0000-dependency-management-phase-two.md

```diff
@@ -0,0 +1,201 @@
+The new process for managing dependencies will use the `buildpack.toml` as a
+source of "truth" for the latest versions we support, instead of using a
+separate `known-versions.json` file. Metadata will also be generated and
+updated in place with a pull request to the `buildpack.toml`, rather than being
+pushed to an intermediary `metadata.json` file. The overall process will be
+closer to what someone would do if they were updating the dependencies
+themselves. The overall steps will be:
+
+1. Pick up all versions of the dependency, using code from Phase 1 in
+  `<buildpack>/dependency/retrieval`
+2. Compare the list of all versions with the version constraints and supported
+  versions in the `buildpack.toml` to determine what versions can be updated.
+3. Generate metadata for each of the versions to be added using code from Phase
+  one in `<buildpack>/dependency/metadata`.
+4. If the `URI` and `SHA256` are missing from the metadata, the code will be
+  compiled with the code from `<buildpack>/dependency/compilation`.
+  - Test the compiled dependency using the test from `<buildpack>/dependency/test`
+  - Upload the dependency to the dependency GCP bucket
+  - Add the GCP bucket access URI and the SHA256 of the dependency to the
+    metadta `URI` and `SHA256` fields
+5. Update the `buildpack.toml` with the new versions and metadata
+
+This new plan eliminates extra steps of storing metadata in a file for later
+use, and then separately updating the dependencies on a timer.
+
+### Workflows and Actions
+Pending the Phase 1 RFC is accepted, and it's decided dependency logic will
+live alongside the buildpack, workflows will be rolled out to all of the
+buildpacks from
+[paketo-buildpacks/github-config](https://github.com/paketo-buildpacks/github-config)
+inside the `implementation/.github/workflows` location.
+
+Actions will live in the github-config repository as well, under a
+directory named `dependency`, inside of the `actions` directory.
+
+The github-config repository `CODEOWNERS` file will be updated to set the
+`@paketo-buildpacks/dependencies-maintainers` to the owners of the
+`actions/dependency` and `implementation/.github/workflows/<all
+dependency-related workflows>`.
+
+
+Each of the step s outlined in the overview will be converted into workflow
+steps.
+
+Workflow 1:
+- Picking up new versions with buildpack code will be a workflow step
+  It should run twice per day.
+- Figuring out which versions will be added to the `buildpack.toml` will be
+  rolled into an action that outputs versions.
+- A Github Actions [`matrix
+  strategy`](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy)
+  will be used to iterate over each version to be added, to generate metadata
+  using the code from within the buildpacks.
+- In the same matrix loop, dependency compilation will be kicked off with the
+  code from the buildpacks on the basis of whether the metadata contains the
```

Comment:
Is the logic to detect if we need to compile or not going to be in github actions configuration or in an executable in the buildpack?

Phrased another way - if you were running the executable locally would you expect the user to know whether to compile (i.e. the logic lives in the action config) or the executable will decide for them (i.e. the logic is in the executable in the buildpack)

Reply:
It'll be in the buildpack code, outlined in Phase 1 RFC. A user does not need to know this information, I would expect the buildpack code to contain this logic